packages/runtime/src/state/makeswift-api-client.ts

```diff
@@ -1,44 +1,98 @@
+import * as LocalizedResourcesMap from './modules/localized-resources-map'
+import { type Action, ActionTypes, apiResourceFulfilled, setLocalizedResourceId } from './actions'
 import {
-  APIResource,
   APIResourceType,
-  Swatch,
-  File,
-  Typography,
-  GlobalElement,
-  PagePathnameSlice,
-  Table,
-  LocalizedGlobalElement,
+  type APIResource,
+  type Swatch,
+  type File,
+  type Typography,
+  type GlobalElement,
+  type PagePathnameSlice,
+  type Table,
+  type LocalizedGlobalElement,
+  type APIResourceLocale,
 } from '../api'
 
-const reducer = APIResources.reducer
+const reducer = combineReducers({
+  apiResources: APIResources.reducer,
+  localizedResourcesMap: LocalizedResourcesMap.reducer,
+})
 
 export type State = ReturnType<typeof reducer>
 
-export type SerializedState = APIResources.SerializedState
+export type SerializedState = {
+  apiResources: APIResources.SerializedState
+  localizedResourcesMap: LocalizedResourcesMap.SerializedState
+}
 
-export function getSerializedState(state: State): SerializedState {
-  return APIResources.getSerializedState(state)
+function getLocalizedResourceId(
+  state: State,
+  locale: string,
+  resourceId: string,
+): string | undefined | null {
+  return LocalizedResourcesMap.getLocalizedResourceId(
+    state.localizedResourcesMap,
+    locale,
+    resourceId,
+  )
 }
 
-export function getHasAPIResource(
+export function getHasAPIResource<T extends APIResourceType>(
   state: State,
   resourceType: APIResourceType,
   resourceId: string,
+  locale?: APIResourceLocale<T>,
 ): boolean {
-  return APIResources.getHasAPIResource(state, resourceType, resourceId)
+  switch (resourceType) {
+    case APIResourceType.LocalizedGlobalElement:
+      if (locale == null) {
+        console.error(`Attempt to access ${resourceType} ${resourceId} without a locale`)
+        return false
+      }
+
+      const localizedId = getLocalizedResourceId(state, locale, resourceId)
+      return (
+        localizedId != null &&
+        APIResources.getHasAPIResource(state.apiResources, resourceType, localizedId, locale)
+      )
+
+    default:
+      return APIResources.getHasAPIResource(state.apiResources, resourceType, resourceId, locale)
+  }
 }
 
 export function getAPIResource<T extends APIResourceType>(
   state: State,
   resourceType: T,
   resourceId: string,
+  locale?: APIResourceLocale<T>,
 ): Extract<APIResource, { __typename: T }> | null {
-  return APIResources.getAPIResource(state, resourceType, resourceId)
+  switch (resourceType) {
+    case APIResourceType.LocalizedGlobalElement:
+      if (locale == null) {
+        console.error(`Attempt to access ${resourceType} ${resourceId} without a locale`)
+        return null
+      }
+
+      const localizedId = getLocalizedResourceId(state, locale, resourceId)
+      return localizedId != null
+        ? APIResources.getAPIResource(state.apiResources, resourceType, localizedId, locale)
+        : null
```

Comment:
Global to localized element ID translation logic, moved from `MakeswiftHostApiClient`: https://github.com/makeswift/makeswift/pull/836/files#diff-e66ba2364c981fb13b774e6d7484c9ed1599cc9131c664ec58d4c64db6eb9268R114